pkgs/applications/window-managers/river/default.nix

```diff
@@ -24,62 +12,33 @@ stdenv.mkDerivation rec {
     owner = "ifreund";
     repo = pname;
     rev = "0c8e718d95a6a621b9cba0caa9158915e567b076";
-    sha256 = "sha256-c3lzsA2oml7DlfYA5mipHafF3Y3mqY1eJR6e2H8DUMo=";
+    sha256 = "1jjh0dzxi7hy4mg8vag6ipfwb9qxm5lfc07njp1mx6m81nq76ybk";
     fetchSubmodules = true;
   };
 
-  nativeBuildInputs = [
-    pkg-config
-    scdoc
-    wayland-protocols
-    zig
+  buildInputs = [ xwayland wayland-protocols wlroots pixman
+    libxkbcommon pixman udev libevdev libX11 libGL
   ];
-  buildInputs = [
-    libGL
-    libX11
-    libevdev
-    libxkbcommon
-    pixman
-    pixman
-    udev
-    wayland
-    wlroots
-    xwayland
-  ];
-
-  dontConfigure = true;
 
-  buildPhase = ''
-    runHook preBuild
+  preBuild = ''
     export HOME=$TMPDIR
-    zig build -Dman-pages -Drelease-safe -Dxwayland --prefix $out
-    runHook postBuild
   '';
-
   installPhase = ''
-    runHook preInstall
-    zig build -Dman-pages -Drelease-safe -Dxwayland --prefix $out install
-    runHook postInstall
+    zig build -Drelease-safe -Dxwayland -Dman-pages --prefix $out install
   '';
 
+  nativeBuildInputs = [ zig wayland scdoc pkg-config ];
+
+  installFlags = [ "DESTDIR=$(out)" ];
```

Comment:
It should not be needed. DESTDIR has [the same semantics](https://github.com/ziglang/zig/issues/2929) of autotools' DESTDIR variable; therefore, it is not useful for Nix. 

DESTDIR is useful for classic package managers/build scripts like Slackbuilds or PKGBUILDs.